setup-build-env.sh

```diff
@@ -4,57 +4,56 @@ ensure_lxd() {
-  local DOTNET_SDK="${DOTNET_SDK:-dotnet-sdk-7.0.100-linux-ppc64le.tar.gz}"
-  local PATCH_FILE="${PATCH_FILE:-runner-ppc64le.patch}"
+  local BUILD_PREREQS_PATH="${SRCDIR}/build-files"
+  if [ ! -d "${BUILD_PREREQS_PATH}" ]; then
+      echo "Check the BUILD_PREREQS_PATH specification" >&2
+      exit 3
+  fi
+  local PATCH_FILE="${PATCH_FILE:-runner-${ARCH}.patch}"
 
   local BUILD_CONTAINER
   BUILD_CONTAINER="gha-builder-$(date +%s)"
-  lxc launch "${OS_NAME}:${OS_VERSION}" "${BUILD_CONTAINER}" 
+  lxc launch "${LXD_CONTAINER}" "${BUILD_CONTAINER}" 
   lxc ls
   
   # give container some time to wake up
-  sleep 10
+  sleep 5
   
   echo "Copy the build-image script into gha-builder"
-  lxc file push "${BUILD_PREREQS_PATH}/build-image.sh"  "${BUILD_CONTAINER}/home/ubuntu/"
+  echo ${BUILD_PREREQS_PATH}
+  echo ${BUILD_PREREQS_PATH}/build-image.sh-${ARCH}
+  lxc file push --mode 0755 "${BUILD_PREREQS_PATH}/build-image.sh-${ARCH}" "${BUILD_CONTAINER}${BUILD_HOME}/build-image.sh"
+  sleep 10s
   
-  echo "Copy the dotnet-sdk into gha-builder"
-  lxc file push "${BUILD_PREREQS_PATH}/${DOTNET_SDK}" "${BUILD_CONTAINER}/home/ubuntu/"
+  if [ -n "${DOTNET_SDK}" ]; then
+      echo "Copy the dotnet-sdk into gha-builder"
+      lxc file push ${BUILD_PREREQS_PATH}/${DOTNET_SDK} "${BUILD_CONTAINER}${BUILD_HOME}"
+  fi
   
   echo "Copy the patch file into gha-builder"
-  lxc file push "${BUILD_PREREQS_PATH}/${PATCH_FILE}" "${BUILD_CONTAINER}/home/ubuntu/"
+  lxc file push ${BUILD_PREREQS_PATH}/${PATCH_FILE} "${BUILD_CONTAINER}${BUILD_HOME}/"
 
   echo "Copy the register-runner.sh script into gha-builder"
-  lxc file push "${BUILD_PREREQS_PATH}/register-runner.sh" "${BUILD_CONTAINER}/opt/register-runner.sh"
+  lxc file push --mode 0755 ${BUILD_PREREQS_PATH}/register-runner.sh "${BUILD_CONTAINER}/opt/register-runner.sh"
   
   echo "Copy the gha-service unit file into gha-builder"
-  lxc file push "${BUILD_PREREQS_PATH}/gha-runner.service" "${BUILD_CONTAINER}/etc/systemd/system/gha-runner.service"
+  lxc file push ${BUILD_PREREQS_PATH}/gha-runner.service "${BUILD_CONTAINER}/etc/systemd/system/gha-runner.service"
 
-  echo "Setting executable permissions on register-runner.sh"
-  lxc exec "${BUILD_CONTAINER}" -- chmod +x /opt/register-runner.sh
- 
-  echo "Setting executable permissions on build-image.sh"
-  lxc exec "${BUILD_CONTAINER}" -- chmod +x /home/ubuntu/build-image.sh
-  
   echo "Running build-image.sh"
-  lxc exec "${BUILD_CONTAINER}" -- /home/ubuntu/build-image.sh
+  lxc exec "${BUILD_CONTAINER}" -- ${BUILD_HOME}/build-image.sh -a ${ACTION_RUNNER}
   
   #TODO: Have better error handling checks
   echo "Runner build complete. Creating image snapshot."
 
-  lxc publish "${BUILD_CONTAINER}" -f --alias "$IMAGE_ALIAS" description="GitHub Actions Ubuntu 20.04 Runner for IBM Power."
+  lxc publish "${BUILD_CONTAINER}" -f --alias "$IMAGE_ALIAS" description="GitHub Actions ${OS_NAME} ${OS_VERSION} for ${ARCH}"
```

Comment:
I *think* the word "Runner" got dropped in this abstraction, and I *think* the word "Runner" has good disambiguation here?

Reply:
"Runner build complete" is to what you are referring? "Runner builder complete"?

Reply:
os_name os_version **Runner** for arch